core/src/main/scala/kafka/controller/ReplicaStateMachine.scala

```diff
@@ -108,11 +109,39 @@ class ZkReplicaStateMachine(config: KafkaConfig,
   override def handleStateChanges(replicas: Seq[PartitionAndReplica], targetState: ReplicaState): Unit = {
     if (replicas.nonEmpty) {
       try {
+        var taskStartMs: Long = Time.SYSTEM.milliseconds
+        val doExtraTiming: Boolean = (targetState == OnlineReplica && replicas.size > 250000)
+        val timings = AggregatedTimings()
```

Comment:
Please see my comments below. With that, this variable can be dropped since we already have timingsOpt.